usr/src/cmd/zpool/zpool_main.c

```diff
@@ -617,6 +623,139 @@ zpool_do_remove(int argc, char **argv)
 }
 
 /*
+ * zpool labelclear <vdev>
+ *
+ * Verifies that the vdev is not active and zeros out the label information
+ * on the device.
+ */
+int
+zpool_do_labelclear(int argc, char **argv)
+{
+	char *vdev, *name = NULL;
+	struct stat st;
+	int c, fd = -1, ret = 0;
+	pool_state_t state;
+	boolean_t inuse = B_FALSE;
+	boolean_t force = B_FALSE;
+
+	/* check options */
+	while ((c = getopt(argc, argv, "f")) != -1) {
+		switch (c) {
+		case 'f':
+			force = B_TRUE;
+			break;
+		default:
+			(void) fprintf(stderr, gettext("invalid option '%c'\n"),
+			    optopt);
+			usage(B_FALSE);
+		}
+	}
+
+	argc -= optind;
+	argv += optind;
+
+	/* get vdev name */
+	if (argc < 1) {
+		(void) fprintf(stderr, gettext("missing vdev device name\n"));
+		usage(B_FALSE);
+	}
+
+	/* Allow bare paths if they exist, otherwise prepend. */
+	if (stat(argv[0], &st) != 0 &&
+	    strncmp(argv[0], ZFS_DISK_ROOTD, strlen(ZFS_DISK_ROOTD)) != 0)
+		ret = asprintf(&vdev, ZFS_DISK_ROOTD "%s", argv[0]);
```

Comment:
printf args should be `"%s%s", ZFS_DISK_ROOTD, argv[0]` so that strange chars in ZFS_DISK_ROOTD can't confuse printf (and so we aren't relying on the macro evaluating to a string literal)